scheduler/pkg/scheduler/filters/serverrequirements.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022 Seldon Technologies Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/seldonio/seldon-core/scheduler/v2/pkg/store"
+)
+
+type ServerRequirementFilter struct{}
+
+func (s ServerRequirementFilter) Name() string {
+	return "ServerRequirementsFilter"
+}
+
+func (s ServerRequirementFilter) Filter(model *store.ModelVersion, server *store.ServerSnapshot) bool {
+	if len(server.Replicas) == 0 {
```

Comment:
We don't know what order the filters are applied in here, and sadly the requirements are not currently stored per server, so it felt necessary here.

For reference, I was thinking of putting in a follow-up PR that would store requirements per server rather than per replica as these should always be the same, and thus it's awkward and duplicative to store this information as we do now.  Would you be happy for that to be handled in a follow-up PR?

Reply:
That would be a nice improvement to "denormalise" it from the replicas to a singular place! Sounds good for a follow-up ticket too.

> We don't know what order the filters are applied in here ...

Why would the order matter, since they're all treated "equally" as being true/false, and any false causes the filter to fail?

Reply:
There's another filter for zero replicas, but within this filter we can't guarantee that it's applied first, hence the use of the guard

Reply:
Ah, okay!